proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +111,335 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
- applies only for the *VectorMatrixMulAdd* operation and is ignored for *VectorMatrixMul* operation. 
+##### Matrix
 
-*Ii* Input Interpretation, *Mi* MatrixInterpretation and *Bi* BiasInterpretation define what type the respective objects
- will be interpreted as. These values are constrained by the combinations allowed by the device, *Matrix* and *Bias*
- are typeless buffers and their respective interpretations determine the types. See [Type Interpretations]
- (#type-interpretations) section for more details.
+The matrix is loaded from the raw-buffer, **matrix resource**,  starting at
+**matrix offset**. The **matrix interpretation** argument specifies the element
+type of the matrix (see [Type Interpretations]). The **matrix M dimension** and
+**matrix K dimension** arguments specify the dimensions of the matrix. The
+**matrix layout** argument specifies the layout of the matrix (see [Matrix
+Layouts]). If the **matrix transpose** is non-zero then the matrix is transposed
+before performing the multiply (see [Matrix Transpose]). For row-major and
+column-major layouts, **matrix stride** specifies the number of bytes to go from
+one row/column to the next.  For optimal layouts, **matrix stride** is ignored.
 
+Only non-packed interpretations are valid for matrices.
 
-*Mi* and *Bi* determines the type of the Weight Matrix and Bias Vector elements.
+The base address of **matrix resource** and **matrix offset** must be 64 byte
+aligned.
 
-For the unpacked case, *M x K* is the dimension of the Matrix, *M* is the size of the result vector, *K* is the size of
-the input vector. For the packed case, the number of components in the input vector must large enough to hold the *K*
-packed components.
 
-*Layout* is an enum value, `DXILMatrixLayout`. Optimal layouts are opaque implementation specific layouts, the D3D call
- `CooperativeVectorConvertMatrix` can be used to convert the *Matrix* to an optimal layout. Row-Major and Column-Major
- layouts are also supported.
 
-The *Transposed* parameter indicates if the *Matrix* is transposed before performing the multiply. In linear algebra,
-the[transpose](https://en.wikipedia.org/wiki/Transpose) of a matrix is an operator which flips a matrix over its
-diagonal; that is, it switches the row and column indices of the matrix. Transposing is not supported for the
-RowMajor/ColumnMajor layouts. Not all component types support transposing. It is left to implementations to define
-which types support matrix transposing. "TransposeSupported" flag from the [CheckFeatureSupport]
-(#check-feature-support) struct is used to determine if a matrix transpose is supported. Note that even for the
-type/interpretation combinations with guaranteed [support](#minimum-support-set), transpose support isn't guaranteed
-and needs to be checked explicitly.
+> TODO: consider using a different set of interpretation values for in-memory
+> interpretations versus vector interpretations.
```

Comment:
Unless there's a strong feeling we should just remove this, maybe I should turn into issue?

Reply:
I am not sure I understand this.

Reply:
Filed https://github.com/microsoft/hlsl-specs/issues/402